### PR TITLE
fixes lavaproof rods, adds to rnd

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Fixes lavaproof rods not working.

Adds them to rnd.

## Why It's Good For The Game

Lavaproof rods working is good.

Them being buildable as a more sensible alternative to the lava staff is good

## Testing

Built with them on lava.

## Changelog
:cl:
add: Adds lavaproof rods to rnd
fix: Fixed lavaproof rods not working on lava
/:cl:

### CODE REVIEW

File: code/game/turfs/simulated/floor/lava.dm [hunk 1]
```diff
@@ -122,6 +122,19 @@
 		else
 			to_chat(user, "<span class='warning'>You need one rod to build a heatproof lattice.</span>")
 			return
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/lava.dm:124]: This isn't your code. but this `return` probably has to be indented one less now that we have a second thing to check in this proc

File: code/game/turfs/simulated/floor/lava.dm [hunk 2]
```diff
@@ -122,6 +122,19 @@
...
+			var/obj/item/stack/tile/plasteel/S = C
+			if(S.use(1))
+				qdel(L)
+				playsound(src, 'sound/weapons/genhit.ogg', 50, 1)
+				to_chat(user, "<span class='notice'>You build a floor.</span>")
+				ChangeTurf(/turf/simulated/floor/plating, keep_icon = FALSE)
+			else
+				to_chat(user, "<span class='warning'>You need one floor tile to build a floor!</span>")
+		else
+			to_chat(user, "<span class='warning'>The plating is going to need some support! Place metal rods first.</span>")
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/lava.dm:138]:
```suggestion
```

**Review by DGamerL — COMMENTED** [review]
Looking really good to me, just one thing that probably can be changed now that the proc has grown a bit

File: code/game/turfs/simulated/floor/lava.dm [hunk 3]
```diff
@@ -121,8 +121,20 @@
 			new /obj/structure/lattice/lava(locate(x, y, z))
 		else
 			to_chat(user, "<span class='warning'>You need one rod to build a heatproof lattice.</span>")
-			return
-
+		return
+	if(istype(C, /obj/item/stack/tile/plasteel))
+		var/obj/structure/lattice/L = locate(/obj/structure/lattice/lava, src)
+		if(L)
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/lava.dm:127]: Last thing, could you make this an early return?

**Review by Sirryan2002 — COMMENTED** [review]
looks good, just address DGamerL's request

File: code/game/turfs/simulated/floor/lava.dm (hunk 3 above)
Comment (Qwertytoforty) [reply code/game/turfs/simulated/floor/lava.dm:127]: sure

File: code/game/turfs/simulated/floor/lava.dm [hunk 4]
```diff
@@ -121,8 +121,21 @@
...
+		var/obj/item/stack/tile/plasteel/S = C
+		if(S.use(1))
+			qdel(L)
+			playsound(src, 'sound/weapons/genhit.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You build a floor.</span>")
+			ChangeTurf(/turf/simulated/floor/plating, keep_icon = FALSE)
+		else
+			to_chat(user, "<span class='warning'>You need one floor tile to build a floor!</span>")
+	else
+		to_chat(user, "<span class='warning'>The plating is going to need some support! Place metal rods first.</span>")
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/lava.dm:138]:
```suggestion
	if(!istype(C, /obj/item/stack/tile/plasteel))
		var/obj/structure/lattice/L = locate(/obj/structure/lattice/lava, src)
		if(!L)
			to_chat(user, "<span class='warning'>The plating is going to need some support! Place metal rods first.</span>")
			return
		var/obj/item/stack/tile/plasteel/S = C
		if(S.use(1))
			qdel(L)
			playsound(src, 'sound/weapons/genhit.ogg', 50, 1)
			to_chat(user, "<span class='notice'>You build a floor.</span>")
			ChangeTurf(/turf/simulated/floor/plating, keep_icon = FALSE)
		else
			to_chat(user, "<span class='warning'>You need one floor tile to build a floor!</span>")
```
Miscommunication, I meant the `if(L)` like this!

File: code/game/turfs/simulated/floor/lava.dm [hunk 5]
```diff
@@ -121,8 +121,20 @@
 			new /obj/structure/lattice/lava(locate(x, y, z))
 		else
 			to_chat(user, "<span class='warning'>You need one rod to build a heatproof lattice.</span>")
+		return
+	if(!istype(C, /obj/item/stack/tile/plasteel))
```
Comment (DGamerL) [inline code/game/turfs/simulated/floor/lava.dm:125]:
```suggestion
	if(istype(C, /obj/item/stack/tile/plasteel))
```
Updated half a second too late, apologies

**Review by DGamerL — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**